packages/core/src/api/webhook.ts

```diff
@@ -0,0 +1,196 @@
+import { makeURLSearchParams, type RawFile, type REST } from '@discordjs/rest';
+import {
+	Routes,
+	type RESTGetAPIChannelMessageResult,
+	type RESTGetAPIWebhookResult,
+	type RESTPatchAPIWebhookJSONBody,
+	type RESTPatchAPIWebhookResult,
+	type RESTPatchAPIWebhookWithTokenMessageJSONBody,
+	type RESTPatchAPIWebhookWithTokenMessageResult,
+	type RESTPostAPIChannelWebhookJSONBody,
+	type RESTPostAPIWebhookWithTokenGitHubQuery,
+	type RESTPostAPIWebhookWithTokenJSONBody,
+	type RESTPostAPIWebhookWithTokenQuery,
+	type RESTPostAPIWebhookWithTokenResult,
+	type RESTPostAPIWebhookWithTokenSlackQuery,
+	type RESTPostAPIWebhookWithTokenWaitResult,
+	type Snowflake,
+} from 'discord-api-types/v10';
+
+export class WebhooksAPI {
+	public constructor(private readonly rest: REST) {}
+
+	/**
+	 * Fetches a webhook
+	 *
+	 * @param id - The id of the webhook
+	 * @param token - The token of the webhook
+	 */
+	public async get(id: Snowflake, token?: string) {
+		return this.rest.get(Routes.webhook(id, token)) as Promise<RESTGetAPIWebhookResult>;
+	}
+
+	/**
+	 * Creates a new webhook
+	 *
+	 * @param channelId - The id of the channel to create the webhook in
+	 * @param data - The data to use when creating the webhook
+	 * @param reason - The reason for creating the webhook
+	 */
+	public async create(channelId: Snowflake, data: RESTPostAPIChannelWebhookJSONBody, reason?: string) {
+		return this.rest.post(Routes.channelWebhooks(channelId), {
+			reason,
+			body: data,
+		}) as Promise<RESTPostAPIWebhookWithTokenResult>;
+	}
+
+	/**
+	 * Edits a webhook
+	 *
+	 * @param id - The id of the webhook to edit
+	 * @param webhook - The new webhook data
+	 * @param token - The token of the webhook
+	 * @param reason - The reason for editing the webhook
+	 */
+	public async edit(id: Snowflake, webhook: RESTPatchAPIWebhookJSONBody, token?: string, reason?: string) {
+		return this.rest.patch(Routes.webhook(id, token), { reason, body: webhook }) as Promise<RESTPatchAPIWebhookResult>;
+	}
+
+	/**
+	 * Deletes a webhook
+	 *
+	 * @param id - The id of the webhook to delete
+	 * @param token - The token of the webhook
+	 * @param reason - The reason for deleting the webhook
+	 */
+	public async delete(id: Snowflake, token?: string, reason?: string) {
```

Comment:
`token` and `reason` should be consolidated into an object since they're both optional.